src/snowflake/snowpark/modin/pandas/dataframe.py

```diff
@@ -254,9 +254,9 @@ def __repr__(self):
         str
         """
         # TODO: SNOW-1063346: Modin upgrade - modin.pandas.DataFrame functions
-        num_rows = pandas.get_option("display.max_rows") or 10
+        num_rows = pandas.get_option("display.max_rows") or len(self.index)
```

Comment:
`len(self.index)` can be pretty expensive since it still pulls the index (at least until we have a cache for the length), I think keeping the hardcoded "or 10" here and differing from pandas in this case is perhaps a better call

Reply:
I did consider what you suggested, but the reason I chose to go with `len(self.index)` is because:
- User must opt-in to this behavior by setting `display.max_rows = None`
- Users have no option outside of converting the entire dataset to pandas in order to view all of their rows
- We kind of end up treating users as though they are not capable of making the decision themselves to pull out all of the rows.

I think a better way forward would be to throw a warning when `display.max_rows` is set to `None` to let the user know that would be pulling out all of the data. Then at least they are informed when they set it.

We should also not require `len(self.index)` to pull the index out of the database. That needs to be fixed separately.

Reply:
That makes sense to me -- going to approve the PR, we can add the warning in this PR or in a follow-up good with either option.

Reply:
Can we just use `len(self)`? It will make a count query which is better than pulling the data locally. At least we don't need to worry about OOM if data is huge.